packages/rum-recorder/src/domain/rrweb/record.spec.ts

```diff
@@ -1,40 +1,296 @@
 import { createNewEvent, isIE } from '@datadog/browser-core'
-import { RecordType, Record } from '../../types'
+import { serializedNodeWithId, NodeType } from 'rrweb-snapshot'
+import { collectAsyncCalls } from '../../../test/utils'
+import { Record, RecordType, IncrementalSource, MutationData, FullSnapshotRecord } from '../../types'
 import { record } from './record'
 
+// Each full snapshot is generating two records, a Meta record and a FullSnapshot record
+const RECORDS_PER_FULL_SNAPSHOTS = 2
+
 describe('record', () => {
+  let sandbox: HTMLElement
   let input: HTMLInputElement
   let stop: (() => void) | undefined
+  let emitSpy: jasmine.Spy<(record: Record) => void>
+  let waitEmitCalls: (expectedCallsCount: number, callback: () => void) => void
+  let expectNoExtraEmitCalls: (done: () => void) => void
 
   beforeEach(() => {
     if (isIE()) {
       pending('IE not supported')
     }
 
+    emitSpy = jasmine.createSpy()
+    ;({ waitAsyncCalls: waitEmitCalls, expectNoExtraAsyncCall: expectNoExtraEmitCalls } = collectAsyncCalls(emitSpy))
+    sandbox = document.createElement('div')
+    sandbox.id = 'sandbox'
     input = document.createElement('input')
-    document.body.appendChild(input)
+    sandbox.appendChild(input)
+    document.body.appendChild(sandbox)
```

Comment:
nit: this could be extracted to a function in order to have the same level of abstraction than other things happening in beforeEach